api/v1beta2/cluster_types.go

```diff
@@ -536,22 +536,29 @@ type ClusterAvailabilityGate struct {
 	Polarity ConditionPolarity `json:"polarity,omitempty"`
 }
 
-// Topology encapsulates the information of the managed resources.
-type Topology struct {
-	// class is the name of the ClusterClass object to create the topology.
+// ClusterClassRef is the ref to the ClusterClass that should be used for the topology.
+type ClusterClassRef struct {
+	// name is the name of the ClusterClass that should be used for the topology.
 	// +required
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:MaxLength=253
-	Class string `json:"class"`
+	Name string `json:"name"`
 
-	// classNamespace is the namespace of the ClusterClass object to create the topology.
+	// namespace is the namespace of the ClusterClass that should be used for the topology.
 	// If the namespace is empty or not set, it is defaulted to the namespace of the cluster object.
 	// Value must follow the DNS1123Subdomain syntax.
 	// +optional
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:MaxLength=253
 	// +kubebuilder:validation:Pattern=`^[a-z0-9](?:[-a-z0-9]*[a-z0-9])?(?:\.[a-z0-9](?:[-a-z0-9]*[a-z0-9])?)*$`
-	ClassNamespace string `json:"classNamespace,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// Topology encapsulates the information of the managed resources.
+type Topology struct {
```

Comment:
We have to decide what we want to do with the builtin variable.

I would suggest to introduce the classRef there in the same way and deprecate the existing class + classNamespace fields there (note to myself: also have to update patch_validation.go)

Reply:
cc @fabriziopandini @chrischdi

Reply:
I'm ok with the proposed approach (add new classRef var, deprecate old standalone vars)